app/src/androidTest/java/org/wikipedia/search/SearchIntentTest.kt

```diff
@@ -0,0 +1,113 @@
+package org.wikipedia.search
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.Matchers.allOf
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.wikipedia.Constants
+import org.wikipedia.R
+import org.wikipedia.TestUtil
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class SearchIntentTest {
+
+    @Rule
+    @JvmField
+    var mActivityTestRule = ActivityScenarioRule<SearchActivity>(SearchActivity.newIntent(ApplicationProvider.getApplicationContext(),
+            Constants.InvokeSource.INTENT_SHARE, "barack obama"))
+
+    @Test
+    fun testSearchActivityWithQuery() {
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        TestUtil.delay(5)
+
+        onView(allOf(withId(R.id.page_list_item_title), withText("Barack Obama"), isDisplayed()))
+                .check(matches(withText("Barack Obama")))
+
+        TestUtil.delay(2)
+
+        device.setOrientationRight()
+        TestUtil.delay(2)
+
+        Espresso.pressBack()
+        TestUtil.delay(1)
+
+        onView(allOf(withId(R.id.page_list_item_title), withText("Barack Obama"), isDisplayed()))
+                .check(matches(withText("Barack Obama")))
+
+        device.setOrientationNatural()
+        device.unfreezeRotation()
+
+        TestUtil.delay(2)
+
+        TestUtil.setAirplaneMode(true)
```

Comment:
On my Samsung device, this step will block the test and will not go back to the app. (but I also guess it should be fine if runs on an emulator).